modules/mobile/app/controllers/mobile/v1/messages_controller.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Mobile
+  module V1
+    class MessagesController < MessagingController
+      include Filterable
+
+      def thread
+        message_id = params[:id].try(:to_i)
```

Comment:
I know we're primarily just trying to imitate the My_health version of this and this is how they do it but it's weird to me that it's trying to turn the id into a integer only for it to get put back into a string.

Reply:
I agree with Andrew. I assume the "try" is intended to catch any non-integer based string. I don't think it's possible for a path param to be nil, but I could be mistaken about that. If it's a non-integer string, like "foo", it will just be converted to zero. And the upstream request with either "foo" or 0 will still be 404 so there's no meaningful difference. I think the only things we could do differently are maybe explicitly raise not found if the param is nil or if `params[:id].try(:to_i)` is zero. Either way, I don't think this `to_i` helps u.

Reply:
I copies it from the controller mentioned in the discovery and the best I could tell was they were guarding against something I was missing. I agree with both of you though, ultimately it would 404 based on failure cases